python/ray/data/dataset.py

```diff
@@ -2123,6 +2127,7 @@ def take(self, limit: int = 20) -> List[T]:
             output.append(row)
             if len(output) >= limit:
                 break
+        self._shutdown_current_executor()
```

Comment:
Should this shutdown be called for each consumption API (directly or indirectly)? Not clear why only tak/take_all are affected.

Reply:
I think the difference is sync vs async consumption APIs. We only want to shut them down for "sync" APIs like show()/take() which block and return to the console. For non-blocking ones, we don't want to shutdown.

Reply:
Could we add this to shutdown comments?
We have APIs like `to_pandas()` which are also taking all blocks instead of returning an iterator, so might be affected as well.

Reply:
Renamed and added comments.